mlflow/models/utils.py

```diff
@@ -257,41 +276,78 @@ def _read_example(mlflow_model: Model, path: str):
 
     :param mlflow_model: Model metadata.
     :param path: Path to the model directory.
-    :return: Input example or None if the model has no example.
+    :return: Input example data or None if the model has no example.
     """
     if mlflow_model.saved_input_example_info is None:
         return None
     example_type = mlflow_model.saved_input_example_info["type"]
     if example_type not in ["dataframe", "ndarray", "sparse_matrix_csc", "sparse_matrix_csr"]:
         raise MlflowException(f"This version of mlflow can not load example of type {example_type}")
     input_schema = mlflow_model.signature.inputs if mlflow_model.signature is not None else None
-    path = os.path.join(path, mlflow_model.saved_input_example_info["artifact_path"])
+    path = os.path.join(path, mlflow_model.saved_input_example_info[INPUT_EXAMPLE_PATH])
     if example_type == "ndarray":
         return _read_tensor_input_from_json(path, schema=input_schema)
-    elif example_type in ["sparse_matrix_csc", "sparse_matrix_csr"]:
+    if example_type in ["sparse_matrix_csc", "sparse_matrix_csr"]:
         return _read_sparse_matrix_from_json(path, example_type)
-    else:
-        return dataframe_from_raw_json(path, schema=input_schema)
+    return dataframe_from_raw_json(path, schema=input_schema)
+
+
+def _save_params_example(mlflow_model: Model, params_example: Optional[Dict[str, Any]], path: str):
+    """
+    Save params_example to a file on the given path and updates passed Model with example metadata.
+
+    :param mlflow_model: Model metadata that will get updated with the example metadata.
+    :param params_example: Optional params dictionary that PyFunc wrapped models can use for scoring.
+    :param path: Where to store the example file. Should be model the model directory.
+    """
+    if params_example is not None:
+        try:
+            _infer_param_schema(params_example)
+        except MlflowException:
+            _logger.warning(f"Invalid params found in input example: {params_example}")
+            raise
```

Comment:
What error message does a user receive here?

Reply:
Whatever possible error messages in _infer_param_schema, normally it's `Expected parameters to be dict, got ...` or `"Failed to infer schema for parameters: {invalid_params}"`. Add this catch to print out the warning as this comes from params_example.